content/en/blog/2023/basic-connector/index.md

```diff
@@ -0,0 +1,574 @@
+                 // OpenTelemetry connector factory to make a factory for connectors
+                 return connector.NewFactory(
+                 typeStr,
+                 createDefaultConfig,
+                 connector.WithTracesToMetrics(createTracesToMetricsConnector, component.StabilityLevelAlpha))
+             }
+
+    &nbsp;<br>
+
+factory.go file once finished:
+
+    package exampleconnector
+
+    import (
+        "context"
+
+        "go.opentelemetry.io/collector/component"
+        "go.opentelemetry.io/collector/connector"
+        "go.opentelemetry.io/collector/consumer"
+    )
+
+    const (
+        defaultVal = "request.n"
+        // this is the name used to refer to the connector in the config.yaml
+        typeStr = "example"
+    )
+
+
+    // NewFactory creates a factory for tailtracer connector.
+    func NewFactory() connector.Factory {
+        // OpenTelemetry connector factory to make a factory for connectors
+        return connector.NewFactory(
+        typeStr,
+        createDefaultConfig,
+        connector.WithTracesToMetrics(createTracesToMetricsConnector, component.StabilityLevelAlpha))
+    }
+
+
+    func createDefaultConfig() component.Config {
+        return &Config{
+            AttributeName: defaultVal,
+        }
+    }
+
+
+    // defines the consumer type of the connector
+    // we want to consume traces and export metrics, therefore, define nextConsumer as metrics, consumer is the next component in the pipeline
+    func createTracesToMetricsConnector(ctx context.Context, params connector.CreateSettings, cfg component.Config, nextConsumer consumer.Metrics) (connector.Traces, error) {
+      c, err := newConnector(params.Logger, cfg)
+      if err != nil {
+        return nil, err
+      }
+      c.metricsConsumer = nextConsumer
+      return c, nil
+    }
+
+&nbsp;<br>
+
+## Implementing the Trace Connector
+
+Implement the methods from the interface component specific to the type of the
+component in the connector.go file. In this tutorial we will implement the
+Traces connector.
+
+1.  Define the connector struct with the desired parameters for your connector
+
+        // schema for connector
+        type connectorImp struct {
+            config Config
+            metricsConsumer consumer.Metrics
+            logger *zap.Logger
+        }
+
+2.  Define the `newConnector` function to create a connector
+
+        // function to create a new connector
+        func newConnector(logger *zap.Logger, config component.Config) (*connectorImp, error) {
+            logger.Info("Building exampleconnector connector")
+            cfg := config.(*Config)
+
+            return &connectorImp{
+              config: *cfg,
+              logger: logger,
+            }, nil
+        }
+
+    The `newConnector` function is a factory function to create an instance of a
+    connector.
+
+3.  Implement `Capabilities` method to properly implement the interface
```

Comment:
Maybe worth commenting what capabilities are and why we return this specific value here? (i.e. talk about the component not modifying the `pdata` structs it handles)